master/buildbot/status/buildrequest.py

```diff
@@ -165,6 +170,10 @@ def asDict(self):
     def asDict_async(self, request=None):
         result = {}
 
+        builder = self.status.getBuilder(self.getBuilderName())
+        if not builder:
+            defer.returnValue(result)
+            return
```

Comment:
I'm not sure if there is "good" time where we have no builder associated but... if we did then wouldn't this return mean that we won't get some of the info like source stamps?